tests/php/tests/includes/test_class.access-token.php

```diff
@@ -0,0 +1,33 @@
+<?php
+
+namespace WP_Job_Manager;
+
+use WP_UnitTestCase;
+
+class WP_Test_Access_Token extends WP_UnitTestCase {
+
+	public function test_correct_token_verified_succesfully() {
+		$access_token = new Access_Token( [ 'user_id' => 10 ] );
+		$token = $access_token->create();
+
+		self::assertTrue( ( new Access_Token( [ 'user_id' => 10] ))->verify( $token ) );
```

Comment:
WDYT of a static API, `Access_Token::create( $user_id, $expiry )` and `Access_Token::verify( $user_id, $token )`? Is there a benefit in creating instances of the class?

Reply:
So this depends on what you suggest `create` and `verify` are going to be.

If the suggestion is to avoid instances entirely and define static methods everywhere then this is practically the same with adapting a procedural style. If this is the case I think we should switch to using namespaces and simple functions instead of using classes as classes are meant to be used for OOP. On the other hand if we want adopt an OOP style then using contructors and instances is the way to go. There are many new features (e.g. constructor promotion, named arguments etc.) in PHP that suggest it moves towards an OOP style and the sooner we adapt our code the easier it's going to be to use them in the future.

I know that existing code does not follow that and classes are just a collection of staff the relate to something but I think this is a remnant of the early days where namespaces didn't exist and defining classes was the only option to avoid global conflicts.

If on the other hand the suggestion is to use a static method to create the instances themselves, I think it would make sense to do it if we had many and complicated usages of the constructor where in that case defining named constructors would make things easier to read.

Reply:
Yeah looking broadly at code in these plugins in a lot of cases global singletons or just procedural functions seem to be the best fit. For me OOP and creating objects makes sense mainly when we have some state that's kept around and used for multiple method calls or used in multiple ways. For one-off functions that just take some parameters and return a value, we shouldn't complicate things.

And you are right, that's pretty much just namespaced functions. Static classes are still pretty useful for those, having private helper methods and public functions as a clear API surface.

Reply:
In my view, global singletons are never the best fit and procedural functions would not be my choice in most cases but this I think is part of a bigger discussion.

What I am trying to say is that if we should use the appropriate language constructs depending on the style of the approach, namespaces and functions for procedural and instances for OOP.